lib/dep-graph-builders/npm-lock-v2/index.ts

```diff
@@ -379,3 +401,34 @@ export const getChildNodeKey = (
 
   return undefined;
 };
+const checkOverrides = (
+  ancestry: Ancestry[],
+  overrides: Overrides,
+): string | undefined => {
+  const ancestryWithoutRoot = ancestry.slice(1);
+  // First traverse into overrides from root down
+  for (const [idx, pkg] of ancestryWithoutRoot.entries()) {
+    // Do we have this in overrides
+    const value =
+      overrides[pkg.name] || overrides[`${pkg.name}@${pkg.version}`];
+
+    if (value) {
+      if (typeof value !== 'string') {
+        // Check for dotted resolution
+        if (idx + 1 === ancestryWithoutRoot.length && value['.']) {
+          return value['.'];
+        }
```

Comment:
Needs explanation

Reply:
Resolutions can be written as a dot inside the object with its key as name.

To resolve this I have to know 
1. Are we looking at the package that we actually want to find a resolution for - this is the index check, we need to know we are at the last index of the ancestry array.
2. I then need to make sure the ['.'] value even exists.